### ISSUE TITLE
Camera: Rotation on pitch axis is buggy and avatar direction vector is not properly considered when NavigationInfo type is Walking or Flying or Examining

### ISSUE THREAD

**Issue by RealityDaemon** [RealityDaemon/x3d-finely-sharpened#8]

**Comment by RealityDaemon** [comment]
High severity

**Comment by RealityDaemon** [comment]
Fixes applied to test camera implementation
Used this example for testing: /x3d-examples/SwitchAndEventModel/skybox-switching-with-dino.x3d

**Comment by RealityDaemon** [comment]
In the function that translates Euler angles (pitch yaw and roll) 
To make an Orientation Quaternion - that is a Quaternion that handles rotations well I had to perform the following steps:

1) Translate the euler angles to Q.xyzw 
2) Normalize the output Quaternion
3) Conjugate what was normalized in step 2
4) then normalize again to produce the desired orientation Quaternion.

For the complete solution to this bug fix I had to invert camera_pitch, camera_yaw, camera_roll angles also ensuring they are in radians. 
Furthermore for Walk(), Strafe(), and Fly() Camera operations I had to make sure the camera position was modified using the correct Look direction vector - essentially the camera has to move in the Look direction the user is facing.

To get the correct Look vector direction based on the current Orientation Quaternion I had to perform the below calculation:
Vector3 lookat = QuaternionExtensions.Rotate(Orientation, Vector3.UnitZ);
